deltacat/io/dataset.py

```diff
@@ -9,7 +9,7 @@
 from ray.data import Dataset
 from ray.data.datasource import DefaultBlockWritePathProvider, \
     BlockWritePathProvider
-from ray.data.impl.arrow_block import T
+from ray.data._internal.arrow_block import T
```

Comment:
You can just create a new TypeVar for generics instead of importing from ray _internal module.

Reply:
Yeah, probably a good idea to reduce as many imports from `_internal` as we can